three-cards-divination/src/test/java/com/cyber/mysticism/tarot/controller/ThreeCardsDivinationControllerTest.java

```diff
@@ -22,19 +25,25 @@ class ThreeCardsDivinationControllerTest {
     private ThreeCardsDivinationController threeCardsDivinationController;
 
     @Test
-    void performReading() throws Exception {
+    void performReading_authorized() throws Exception {
         Card card = new Card("The Fool", 0, "Major Arcana", "m00.jpg", List.of("Watch for new projects and new beginnings"));
         Map<String, Card> extracted = Map.of("past", card);
-        when(threeCardsDivinationService.getReading()).thenReturn(extracted);
+        when(threeCardsDivinationService.getReadingForUser("user", "email")).thenReturn(extracted);
 
-        assertEquals(extracted, threeCardsDivinationController.performReading());
+        assertEquals(extracted, threeCardsDivinationController.performReadingForUser("Basic dXNlcjplbWFpbA=="));
     }
```

Comment:
**Suggestion:** Avoid hardcoding the authorization string in your tests. Instead, you can create a constant or a method to generate it. [best practice]
```suggestion
    private static final String AUTHORIZATION_STRING = "Basic dXNlcjplbWFpbA==";
    ...
    void performReading_authorized() throws Exception {
        ...
        assertEquals(extracted, threeCardsDivinationController.performReadingForUser(AUTHORIZATION_STRING));
    }
```